.github/workflows/ci.yml

```diff
@@ -0,0 +1,104 @@
+name: CI Pipeline
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+ 
+concurrency:
+  group: ci-${{ github.head_ref || github.ref || github.run_id }}
+  cancel-in-progress: true
+ 
+jobs:
+  validate:
```

Comment:
I think we can find a better name, "validate" is a big vague. Why not "full-ci-pipeline" or something? (I am opened to suggestions)